Flagsmith.FlagsmithClient/PollingManager.cs

```diff
@@ -21,19 +23,23 @@ public PollingManager(Func<Task> callBack, int intervalSeconds = 10)
             _Interval = intervalSeconds * 1000; //convert to milliseconds
         }
         /// <summary>
-        /// Start calling callback continously after provided interval
+        /// Start calling callback continuously after provided interval
         /// </summary>
         /// <returns>Task</returns>
         public async Task StartPoll()
         {
+            // Force a first call of the callback at least once and synchronously if it is awaited.
+            await _CallBack.Invoke();
             _CancellationTokenSource.Token.ThrowIfCancellationRequested();
-            while (true)
+            _timer = new Timer(async (object state) =>
             {
-                await _CallBack.Invoke();
-                await Task.Delay(_Interval, _CancellationTokenSource.Token);
                 if (_CancellationTokenSource.Token.IsCancellationRequested)
-                    break;
-            }
+                {
+                    _timer.Dispose();
+                    return;
+                }
+                await _CallBack.Invoke();
+            }, null, _Interval, _Interval);
```

Comment:
From what I can tell, [this](https://learn.microsoft.com/en-us/dotnet/api/system.threading.timer.-ctor?view=net-7.0#system-threading-timer-ctor(system-threading-timercallback-system-object-system-timespan-system-timespan)) is the relevant constructor for this Timer instance? It look me a bit of time to locate that though as it seems that the most common example of the `Timer` class is from `System.Timers` which only has 0/1 argument constructors. 

Perhaps Visual Studio / VSCode has some form of intellisense on this so a comment isn't necessary but I just wanted to highlight that it took me some time to find it - because I wasn't sure why we were providing the `_Interval` argument twice.

Reply:
Well, it was easy for me to find this constructor, maybe, because I was looking for this particular constructor signature, since I wanted to execute the callback synchronously a first time, and then use the Timer to execute the callback every _Interval time, but delaying the first execution the same _Interval time to not repeat the first call.